float64 xCen
float64 yCen